src/nvim/diff.c

```diff
@@ -2017,17 +2017,24 @@ int diff_infold(win_T *wp, linenr_T lnum)
 }
 
 /// "dp" and "do" commands.
-///
-/// @param put
-void nv_diffgetput(int put)
+void nv_diffgetput(bool put, size_t count)
 {
   exarg_T ea;
-  ea.arg = (char_u *)"";
+  char buf[30];
+
+  if (count == 0) {
+    ea.arg = (char_u *)"";
+  } else {
+    vim_snprintf(buf, 30, "%ld", count);
```

Comment:
Thanks for that link. 

[This link](http://stackoverflow.com/a/3168298/152142) claims `%z` is C99. Since size_t is unsigned perhaps we should use `%zu`. 

[This MSDN page](http://support.microsoft.com/kb/2967191) says (click "expand all" and search for C99):

> The printf and scanf families of functions now support the new C99 format string improvements.

Perhaps someone with VS 2015 CTP can check the inttypes.h (or equivalent) header.